WalletWasabi/Blockchain/Transactions/TransactionHistoryBuilder.cs

```diff
@@ -77,12 +80,36 @@ public List<TransactionSummary> BuildHistorySummary()
 						TransactionId = spenderTxId,
 						BlockIndex = spenderTransaction.BlockIndex,
 						BlockHash = spenderTransaction.BlockHash,
-						IsOwnCoinjoin = spenderTransaction.IsOwnCoinjoin()
+						IsOwnCoinjoin = spenderTransaction.IsOwnCoinjoin(),
+						Inputs = GetInputs(wallet.Network, spenderTransaction),
+						Outputs = GetOutputs(spenderTransaction, wallet.Network),
 					});
 				}
 			}
 		}
 		txRecordList = txRecordList.OrderByBlockchain().ToList();
 		return txRecordList;
 	}
+
+	private static IEnumerable<Output> GetOutputs(SmartTransaction smartTransaction, Network network)
+	{
+		var txOutList = smartTransaction.Transaction.Outputs.Select(
+			txOut =>
+			{
+				var amount = txOut.Value;
+				var address = txOut.ScriptPubKey.GetDestinationAddress(network);
+				var associatedCoin = smartTransaction.WalletOutputs.FirstOrDefault(smartCoin => smartCoin.TxOut == txOut);
+				return new Output(amount, address, associatedCoin?.IsSpent() ?? false);
```

Comment:
nit: I'd feel happier if this was a separate method.

As a general rule, if you need to open braces in a lambda it's worth considering making it a separate method.